Core/Src/main.c

```diff
@@ -87,18 +89,36 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+  HAL_TIM_PWM_Start(*htim, TIM_CHANNEL_1); //assuming channel 1 based on schematic "servo header". Timer needs to be started only once in program, so it is outside loop
   while (1)
   {
-    /* USER CODE END WHILE */
+	  int pTxData = 0;
+	  int pRxData = 0;
+	  HAL_Delay(10);
+	  *pTxData = 0b000000011000000000000000;
+			  /*
+			   * 00000001 - first byte sent, contains only start bit.
+			   * second byte: single / diffrential bit set to 1 for single ended. 0's for channel 0, then more 0's for dont care bits.
+			   * third byte: entirely dont care bits, all useful information is sent in the 1 bits.
+			   */
+	  HAL_SPI_TransmitReceive_IT(*hspi, *pTxData, *pRxData, 3); //size = 3 bytes (24bits), Rxdata is pointer to received data, TxData is pointer to transmitted. *hspi has config info
```

Comment:
remember, if you use IT which stands for interrupt, you need to create an interrupt handle to actually take care of the data transfer. It would be much simplier to just the polling/blocking version